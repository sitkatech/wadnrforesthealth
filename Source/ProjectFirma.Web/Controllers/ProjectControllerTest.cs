using ApprovalTests;
using ApprovalTests.Reporters;
using NUnit.Framework;
using ProjectFirma.Web.UnitTestCommon;

namespace ProjectFirma.Web.Controllers
{
    [TestFixture]
    public class ProjectControllerTest : DatabaseDirectAccessTestFixtureBase
    {
        /// <summary>
        ///  **** If this unit test fails, that means a new FK to dbo.Project was added.
        /// Make sure this FK/table is added to the delete statements in the stored procedure 'dbo.pBulkDeleteProjects'
        /// </summary>
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void FlagChangesToForeignKeysReferencingProjectTableForDevelopersToUpdateBulkDeleteProjectsStoredProcedure()
        {
            const string sqlQueryString = @"
                        SELECT  
                            tab1.name AS [table],
                            col1.name AS [column],
                            obj.name AS FK_NAME
                        FROM sys.foreign_key_columns fkc
                        INNER JOIN sys.objects obj
                            ON obj.object_id = fkc.constraint_object_id
                        INNER JOIN sys.tables tab1
                            ON tab1.object_id = fkc.parent_object_id
                        INNER JOIN sys.columns col1
                            ON col1.column_id = parent_column_id AND col1.object_id = tab1.object_id
						INNER JOIN sys.tables tab2
							ON tab2.object_id = fkc.referenced_object_id
						INNER JOIN sys.schemas sch
							ON tab1.schema_id = sch.schema_id
						WHERE
							tab2.name = 'Project' and sch.name = 'dbo'
                        ORDER BY tab1.name;";
            var result = ExecAdHocSql(sqlQueryString);
            Approvals.Verify(result.TableToHumanReadableString());
        }

        /// <summary>
        ///  **** If this unit test fails, that means a new FK to dbo.ProjectUpdateBatch was added.
        /// Make sure this FK/table is added to the delete statements in the stored procedure 'dbo.pBulkDeleteProjects'
        /// </summary>
        [Test]
        [UseReporter(typeof(DiffReporter))]
        public void FlagChangesToForeignKeysReferencingProjectUpdateBatchTableForDevelopersToUpdateBulkDeleteProjectsStoredProcedure()
        {
            const string sqlQueryString = @"
                        SELECT  
                            tab1.name AS [table],
                            col1.name AS [column],
                            obj.name AS FK_NAME
                        FROM sys.foreign_key_columns fkc
                        INNER JOIN sys.objects obj
                            ON obj.object_id = fkc.constraint_object_id
                        INNER JOIN sys.tables tab1
                            ON tab1.object_id = fkc.parent_object_id
                        INNER JOIN sys.columns col1
                            ON col1.column_id = parent_column_id AND col1.object_id = tab1.object_id
						INNER JOIN sys.tables tab2
							ON tab2.object_id = fkc.referenced_object_id
						INNER JOIN sys.schemas sch
							ON tab1.schema_id = sch.schema_id
						WHERE
							tab2.name = 'ProjectUpdateBatch' and sch.name = 'dbo'
                        ORDER BY tab1.name;";
            var result = ExecAdHocSql(sqlQueryString);
            Approvals.Verify(result.TableToHumanReadableString());
        }

        [TestFixtureSetUp]
        public void TestFixtureSetup()
        {
            BaseFixtureSetup();
        }

        [TestFixtureTearDown]
        public void TestFixtureTeardown()
        {
            BaseFixtureTeardown();
        }

    }
}
