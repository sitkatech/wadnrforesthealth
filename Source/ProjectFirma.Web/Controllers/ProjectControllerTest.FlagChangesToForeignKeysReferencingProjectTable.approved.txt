table	column	FK_NAME
AgreementProject	ProjectID	FK_AgreementProject_Project_ProjectID
InteractionEventProject	ProjectID	FK_InteractionEventProject_Project_ProjectID
InvoicePaymentRequest	ProjectID	FK_InvoicePaymentRequest_Project_ProjectID
NotificationProject	ProjectID	FK_NotificationProject_Project_ProjectID
ProgramNotificationSentProject	ProjectID	FK_ProgramNotificationSentProject_Project_ProjectID
ProjectClassification	ProjectID	FK_ProjectClassification_Project_ProjectID
ProjectCounty	ProjectID	FK_ProjectCounty_Project_ProjectID
ProjectDocument	ProjectID	FK_ProjectDocument_Project_ProjectID
ProjectExemptReportingYear	ProjectID	FK_ProjectExemptReportingYear_Project_ProjectID
ProjectExternalLink	ProjectID	FK_ProjectExternalLink_Project_ProjectID
ProjectFundingSource	ProjectID	FK_ProjectFundingSource_Project_ProjectID
ProjectGrantAllocationExpenditure	ProjectID	FK_ProjectGrantAllocationExpenditure_Project_ProjectID
ProjectGrantAllocationRequest	ProjectID	FK_ProjectGrantAllocationRequest_Project_ProjectID
ProjectImage	ProjectID	FK_ProjectImage_Project_ProjectID
ProjectImportBlockList	ProjectID	FK_ProjectImportBlockList_Project_ProjectID
ProjectInternalNote	ProjectID	FK_ProjectInternalNote_Project_ProjectID
ProjectLocation	ProjectID	FK_ProjectLocation_Project_ProjectID
ProjectLocationStaging	ProjectID	FK_ProjectLocationStaging_Project_ProjectID
ProjectNote	ProjectID	FK_ProjectNote_Project_ProjectID
ProjectOrganization	ProjectID	FK_ProjectOrganization_Project_ProjectID
ProjectPerson	ProjectID	FK_ProjectPerson_Project_ProjectID
ProjectPriorityLandscape	ProjectID	FK_ProjectPriorityLandscape_Project_ProjectID
ProjectProgram	ProjectID	FK_ProjectProgram_Project_ProjectID
ProjectRegion	ProjectID	FK_ProjectRegion_Project_ProjectID
ProjectTag	ProjectID	FK_ProjectTag_Project_ProjectID
ProjectUpdateBatch	ProjectID	FK_ProjectUpdateBatch_Project_ProjectID
Treatment	ProjectID	FK_Treatment_Project_ProjectID